incubator/hnc/pkg/forest/forest.go

```diff
@@ -45,6 +67,28 @@ func (f *Forest) Unlock() {
 	f.lock.Unlock()
 }
 
+// AddTypeSyncer adds a reconciler to the types list.
+func (f *Forest) AddTypeSyncer(nss TypeSyncer) {
+	f.types = append(f.types, nss)
+}
+
+// HasTypeSyncer returns true if there is already a reconciler with the given GVK.
+func (f *Forest) HasTypeSyncer(gvk schema.GroupVersionKind) bool {
+	for _, t := range f.types {
+		if t.GetGVK() == gvk {
+			return true
+		}
+	}
+	return false
+}
+
+// GetTypeSyncers returns the types list.
```

Comment:
Maybe worth explaining why we're returning a copy instead of the list itself. Eg, this function requires that the mutex is held, but the returned list is safe to use without the mutex being held; also, the caller can modify the returned array without fear of corrupting data.

Reply:
Thanks for the suggestion! Updated the comments. Feel free to take a look.